### ISSUE TITLE
cluster.routing.allocation.enable = None is ignored during rolling restart

### ISSUE THREAD

**Issue by Fizzadar** [elastic/elasticsearch#10493]
We are going through the process of making some changes to our ES cluster (adding new plugins/minor config tweaks). Following the process outlined at:
http://www.elastic.co/guide/en/elasticsearch/guide/master/_rolling_restarts.html, we set `cluster.routing.allocation.enable` to `none` before restarting each node; in the past (<1.4) this has worked a charm, and the shards from the restarted node have remained unassigned until re-enabled. However, at 1.5 this setting is seemingly ignored, and the shards are instantly re-assigned to the remaining nodes in the cluster.

The result is that restarting each node is taking ~30-60 minutes rather than the ~5 it used to take, because we now have to rebalance everything after all the shards are re-assigned, mostly on the wrong nodes.

Any help would be greatly appreciated!

**Comment by rabidscorpio** [comment]
I've also seen something similar with 1.5.0, when I set `cluster.routing.allocation.enable` to `all` after a restart, there's inevitably at least one shard that's relocating and a number of shards that are in initializing state for a while.  At the same time, the shards that became unassigned while the node was restarting take a long time to be reassigned to the node.  What used to take a few mins per node now takes 30-60 mins per node.

**Comment by clintongormley** [comment]
@Fizzadar I've just tried this on 1.5.0. I started 4 nodes, created 4 indices, and then ran this:

    PUT _cluster/settings
    {
      "transient": {
        "cluster.routing.allocation.enable": "none"
      }
    }

Then I killed 3 nodes, which put the cluster into red state, then I restarted them and the cluster went yellow, as expected.  But it only went to green when I reenabled allocation.

What exactly are you doing, and what are you seeing that differs from what you expect?
